### PR TITLE
Add documentation for the VSO-Hash algorithm

### PR DESCRIPTION
**Pull request by erikma**

### CODE REVIEW

File: Documentation/Specs/PagedHash.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Paged File Hashes
+In the BuildXL [cache](../../Public/src/Cache/README.md) the most commonly used file content hash function is not a strict SHA256 of the file content stream, but a paged block hash, labeled as the "VSO-Hash" in the codebase. VSO-Hash originated as part of work within the Visual Studio Online services, which was renamed first to Visual Studio Team Services, then to Azure DevOps.
```
Comment (dannyvv) [inline Documentation/Specs/PagedHash.md]: > Visual Studio Team Serv [](start = 329, length = 24)

I think it was formerly VSTS => VSO => AzDev/ADO (only last week i realized this was not [https://en.wikipedia.org/wiki/ActiveX_Data_Objects](https://en.wikipedia.org/wiki/ActiveX_Data_Objects))

Comment (dannyvv) [reply Documentation/Specs/PagedHash.md]: Data: [https://visualstudio.microsoft.com/vso/](https://visualstudio.microsoft.com/vso/)

---
In reply to: [282247251](https://github.com/microsoft/BuildXL/pull/275#discussion_r282247251) [](ancestors = 282247251)

Comment (dannyvv) [reply Documentation/Specs/PagedHash.md]: I guess we don't kwno the order: [https://www.c-sharpcorner.com/article/vso-is-now-rebranded-as-vsts/](https://www.c-sharpcorner.com/article/vso-is-now-rebranded-as-vsts/)

---
In reply to: [282247431](https://github.com/microsoft/BuildXL/pull/275#discussion_r282247431) [](ancestors = 282247431,282247251)

Comment (erikma) [reply Documentation/Specs/PagedHash.md]: I was going off of the order of announcement emails I got over the past few years :)

Comment (dannyvv) [reply Documentation/Specs/PagedHash.md]: So it was:
VSTS -> VSO -> VSTS -> ADO

---
In reply to: [282247670](https://github.com/microsoft/BuildXL/pull/275#discussion_r282247670) [](ancestors = 282247670,282247431,282247251)

Comment (erikma) [reply Documentation/Specs/PagedHash.md]: I forgot about the original VSTS=="Visual Studio Team System". But I think the VSO paged hash arrived after that.

**Comment by johnterickson** [comment]
Late to the party - was at Build. 

This implementation is ok, but is not that one to recommend:
https://github.com/microsoft/BuildXL/blob/master/Public/Src/Cache/ContentStore/Hashing/VsoHash.cs

The good one uses SIMD (i.e. https://docs.microsoft.com/en-us/windows/desktop/api/bcrypt/nf-bcrypt-bcryptprocessmultioperations).  Those libraries were open source recently but I can't find the right link.

**Pull request merged**